headless-server-commerce-lib/src/main/java/com/coremedia/blueprint/caas/labs/model/CommerceLabsFacade.java

```diff
@@ -143,7 +144,22 @@ public DataFetcherResult<Category> getCategoryByStore(String categoryId, String
   // it is being used by within commerce-schema.graphql as @fetch(from: "@commerceLabsFacade.getCommerceBean(#commerceId, #siteId)")
   @Nullable
   public DataFetcherResult<CommerceBean> getCommerceBean(String commerceId, String siteId) {
-    return fetchData(siteId, connection -> createCommerceBean(commerceId, connection, CommerceBean.class));
+    return fetchData(siteId, connection -> {
+
+      //Check if the id is a product first
+      CommerceId possibleProduct = getProductId(commerceId, connection);
+      CommerceBean bean = connection.getCommerceBeanFactory().createBeanFor(possibleProduct, connection.getInitialStoreContext());
+      try {
+        if (bean != null) {
+          bean.load();
+        }
+      } catch (Exception e) {
+        //It might be a category
+        CommerceId possibleCategory = getCategoryId(commerceId, connection);
+        bean = connection.getCommerceBeanFactory().createBeanFor(possibleCategory, connection.getInitialStoreContext());
+      }
+      return bean;
+    });
```

Comment:
The commerceId is supposed to be a String like "vendor:///catalog/sku/blabla", i think it is misused here to guess a commerce bean for an externalId, right?
The category/product for externalID call should used instead by the client.